### PR TITLE
Populate reset queue with not ready accounts

### PR DESCRIPTION
**Pull request by kddejong**
## Proposed changes

- Switch account reset population to use the new data model 
- Allow for some pagination on populating the account reset

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (changes to code, which do not change application behavior)

## Checklist

- [ ] I have filled out this PR template
- [ ] I have read the [CONTRIBUTING](https://github.com/Optum/dce/blob/master/docs/CONTRIBUTING.md) doc
- [ ] I have added automated tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (`README.md`, inline comments, etc.)
- [ ] I have updated the `CHANGELOG.md` under a `## next` release, with a short summary of my changes

### CODE REVIEW

**Comment by eschwartz** [comment]
@kddejong taking a 👀 now

File: cmd/lambda/populate_reset_queue/main.go [hunk 1]
```diff
@@ -1,80 +1,85 @@
...
+	cfgBldr := &config.ConfigurationBuilder{}
+	settings = &configuration{}
+	if err := cfgBldr.Unmarshal(settings); err != nil {
+		log.Fatalf("Could not load configuration: %s", err.Error())
 	}
 
-	// Get NotReady Accounts
-	accounts, err := dbSvc.FindAccountsByStatus(db.NotReady)
+	// load up the values into the various settings...
+	err := cfgBldr.WithEnv("AWS_CURRENT_REGION", "AWS_CURRENT_REGION", "us-east-1").Build()
```
Comment (eschwartz) [inline cmd/lambda/populate_reset_queue/main.go]: not necessarily a change for this PR, but this is still a little awkward to me how we're pulling in env vars like this, 
which are actually hidden requirements by our DB service.

It seems that if our DB service needs this env var, it's own service constructor should be in charge of declaring that.
I want our controller to only have to declare the env vars it's using directly.

File: cmd/lambda/populate_reset_queue/main.go [hunk 2]
```diff
@@ -1,80 +1,85 @@
...
+
+	services = svcBldr
+
+}
+
+// Handler is the base handler function for the lambda
+func Handler(cloudWatchEvent events.CloudWatchEvent) error {
+
+	var eventSvc eventiface.Servicer
+	if err := services.Config.GetService(&eventSvc); err != nil {
```
Comment (eschwartz) [inline cmd/lambda/populate_reset_queue/main.go]: Didn't we do something in another controller, just like:

```golang
eventSvc := service.GetEventer()
```

that's a nicer pattern, IMO, if we can do it.

Comment (kddejong) [reply cmd/lambda/populate_reset_queue/main.go]: Yea I think we just need to decide how far down that rabbit hole we want to go.  When do we decide if a service is going to have the easier pattern.

File: cmd/lambda/populate_reset_queue/main.go (hunk 1 above)
Comment (kddejong) [reply cmd/lambda/populate_reset_queue/main.go]: I agree with this.  Would be a nice fix.  No repetition if its always needed.

File: pkg/account/service.go [hunk 3]
```diff
@@ -229,6 +230,27 @@ func (a *Service) List(query *Account) (*Accounts, error) {
 	return accounts, nil
 }
 
+// Reset initiates the Reset account process.  It will not change the status as there may
+// be many reasons why a reset is called.  Delete, Lease Ending, etc.
+func (a *Service) Reset(data *Account) error {
```
Comment (eschwartz) [inline pkg/account/service.go]: 👍

**Pull request merged**